clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/StatsAvroGenericDaVinciClient.java

```diff
@@ -33,30 +36,41 @@ public StatsAvroGenericDaVinciClient(AvroGenericDaVinciClient<K, V> delegate, Cl
     if (metricsRepository == null) {
       throw new VeniceClientException("MetricsRepository shouldn't be null");
     }
-    this.clientStatsForSingleGet = BasicClientStats
-        .getClientStats(metricsRepository, clientConfig.getStoreName(), RequestType.SINGLE_GET, clientConfig);
-    this.clientStatsForBatchGet = BasicClientStats
-        .getClientStats(metricsRepository, clientConfig.getStoreName(), RequestType.MULTI_GET, clientConfig);
+    this.clientStatsForSingleGet = BasicClientStats.getClientStats(
+        metricsRepository,
+        clientConfig.getStoreName(),
+        RequestType.SINGLE_GET,
+        clientConfig,
+        VeniceClientType.DA_VINCI_CLIENT);
+    this.clientStatsForBatchGet = BasicClientStats.getClientStats(
+        metricsRepository,
+        clientConfig.getStoreName(),
+        RequestType.MULTI_GET,
+        clientConfig,
+        VeniceClientType.DA_VINCI_CLIENT);
   }
 
   private static <T> CompletableFuture<T> trackRequest(
       BasicClientStats stats,
-      Supplier<CompletableFuture<T>> futureSupplier) {
+      Supplier<CompletableFuture<T>> futureSupplier,
+      int numKeys) {
     long startTimeInNS = System.nanoTime();
     CompletableFuture<T> statFuture = new CompletableFuture<>();
     try {
       return futureSupplier.get().whenComplete((v, throwable) -> {
         if (throwable != null) {
-          stats.recordUnhealthyRequest();
+          stats.emitUnhealthyRequestMetrics(LatencyUtils.getElapsedTimeFromNSToMS(startTimeInNS), numKeys, throwable);
           statFuture.completeExceptionally(throwable);
         } else {
-          stats.recordHealthyRequest();
-          stats.recordHealthyLatency(LatencyUtils.getElapsedTimeFromNSToMS(startTimeInNS));
+          stats.emitHealthyRequestMetrics(LatencyUtils.getElapsedTimeFromNSToMS(startTimeInNS), v);
           statFuture.complete(v);
         }
       });
     } catch (Exception e) {
-      stats.recordUnhealthyRequest();
+      stats.emitUnhealthyRequestMetrics(
+          LatencyUtils.getElapsedTimeFromNSToMS(startTimeInNS),
+          numKeys,
+          HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
DaVinci doesn't do remote calls in the hot path, and if we start emitting the metric with http status, it can cause some confusion.
Is it possible to skip the `http status` dimension for DaVinci tracking?
If it is too challenging to do that, we should add java docs to explain it.

Reply:
Good point. 
A same metric ie: `venice.client.call_count` should have the same set of dimensions always passed in, no optionals as Pre-aggs in the underlying metric store might not support it the right way. TC and FC pass in http status as a dimension and DVC can't. In that case, we either have to come up with a new metric under  `venice.client.' just for DVC as other clients pass in http status or use `venice.da-vinci-client.call_count` prefix as we discussed. What do you think?

Reply:
@FelixGV 
What is your opinion on ^?

DaVinci is more like a local hashmap, so many client-side metrics would hold a slightly different meaning from a remote client.
What will it take to remove `http_status` dimension from DaVinci read metrics while continuing to use the same metric name and `client.type` dimension?
Can we make `http_status` dimension optional and only pass it in for thin-client/fast-client?

Reply:
> Can we make http_status dimension optional and only pass it in for thin-client/fast-client?
One metric should have all dimensions passed in all the time. no optionals as Pre-aggs in the underlying metric storage might not support it the right way. or we have to create different pre-aggs for DVC and use that in grafana, which might be confusing.